lumen/sources/duckdb.py

```diff
@@ -450,6 +483,6 @@ def close(self):
         This method should be called when the source is no longer needed to prevent
         connection leaks and properly clean up server-side resources.
         """
-        if self._connection is not None:
-            self._connection.close()
-            self._connection = None
+        if hasattr(self._local, 'connection') and self._local.connection is not None:
```

Comment:
Surely this should clean up the connection for all threads, not just the local thread.